server/engine/src/main/java/org/eclipse/lsp/cobol/core/engine/dialects/DialectService.java

```diff
@@ -308,11 +312,8 @@ public boolean updateDialects(List<DialectRegistryItem> dialectRegistry) {
             dialectSuppliers.computeIfAbsent(
                 r.getName(),
                 name ->
-                    discoveryService
-                        .loadDialects(r.getUri(), copybookService, messageService)
-                        .stream()
-                        .filter(d -> d.getName().equals(name))
-                        .findFirst()
+                    dialectFactory
```

Comment:
This looks a bit too "enterprise" to me, I suggest just moving the ~10 lines function here.